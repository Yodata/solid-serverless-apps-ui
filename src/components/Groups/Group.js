import React from 'react';
import { connect } from 'react-redux';
import { withStyles } from '@material-ui/core';
// import AppBar from '@material-ui/core/AppBar';
// import Tabs from '@material-ui/core/Tabs';
// import Tab from '@material-ui/core/Tab';
// import StarBorder from '@material-ui/icons/StarBorder';
// import Contacts from '@material-ui/icons/Contacts';
// import ContactMail from '@material-ui/icons/ContactMail';
// import GroupWork from '@material-ui/icons/GroupWork';
// import Button from '@material-ui/core/Button';
import { setGroup } from '../../redux/slices/groupsSlice';
import { signoutUser } from '../../redux/actions/authenticationActions';

/**
 * @Component Group
 * @Description It shows the groups which are available to the user
 */

const styles = theme => ({
    appBar: {
        marginBottom: '2em'
    },
    logout: {
        paddingRight:'10px',
        paddingLeft:'10px',
        marginBottom:'20px',
        textTransform: 'none'
    },
    new: {
        paddingRight:'40px',
        paddingLeft:'40px',
        marginBottom:'20px',
        backgroundColor: theme.palette.new.main,
        textTransform: 'none'
    }
});

export function Group(props) {

    // // const { classes, tabIndex, setTabIndex, logoutUser } = props;
    // const { classes, logoutUser, isAdmin } = props;
    // // const groups = [{ 'Featured': <StarBorder /> }, { 'CRM': <Contacts /> }, { 'Marketing': <ContactMail /> }, { 'Paperless': <GroupWork /> }];
    // // Object.freeze(groups);

    // const handleLogout = () => {
    //     logoutUser();
    // }

    return (
        <React.Fragment>
            {/* <Grid container direction='row' justify='flex-end' alignItems='center'>
                {/* <AppBar position='static' className={classes.appBar} flex='row'>
                <Tabs
                    value={tabIndex}
                    centered
                    onChange={(event, value) => {
                        setTabIndex(value);
                    }}

                >
                    {groups.map(value => (
                        <Tab
                            key={Object.keys(value)[0]}
                            label={Object.keys(value)[0]}
                            icon={Object.values(value)[0]}
                            value={(Object.keys(value)[0]).toLowerCase()} />
                    ))}
                </Tabs> */}
                {/* <Grid item>
                    {isAdmin &&
                        <Button variant='outlined' className={classes.new} onClick={handleLogout}>
                            Add New Application
                        </Button>
                    }
                </Grid>
                <Grid item>
                    <Button className={classes.logout} onClick={handleLogout}>
                        Logout
                    </Button>
                </Grid>
            </Grid>
            </AppBar> */}
        </React.Fragment>
    );
}

const mapStateToProps = state => {
    return {
        tabIndex: state.groups.tabIndex
    };
};

const mapDispatchToProps = dispatch => {
    return {
        setTabIndex: value => dispatch(setGroup(value)),
        logoutUser: () => dispatch(signoutUser())
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(withStyles(styles)(Group));