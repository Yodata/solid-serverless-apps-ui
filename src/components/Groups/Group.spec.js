import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Group } from './Group';

configure({ adapter: new Adapter });

describe("Group Component", () => {
    it("should render the group classification", () => {
        const mockSetTabIndex = jest.fn();
        const wrapper = shallow(<Group classes={{}} setTabIndex={mockSetTabIndex}/>);
        expect(wrapper).toMatchSnapshot();
    });
});