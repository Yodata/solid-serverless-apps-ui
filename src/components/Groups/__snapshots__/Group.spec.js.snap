// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Group Component should render the group classification 1`] = `
<Fragment>
  <WithStyles(ForwardRef(AppBar))
    position="static"
  >
    <WithStyles(ForwardRef(Tabs))
      centered={true}
      onChange={[Function]}
    >
      <WithStyles(ForwardRef(Tab))
        icon={<UNDEFINED />}
        key="Featured"
        label="Featured"
        value="featured"
      />
      <WithStyles(ForwardRef(Tab))
        icon={<UNDEFINED />}
        key="CRM"
        label="CRM"
        value="crm"
      />
      <WithStyles(ForwardRef(Tab))
        icon={<UNDEFINED />}
        key="Marketing"
        label="Marketing"
        value="marketing"
      />
      <WithStyles(ForwardRef(Tab))
        icon={<UNDEFINED />}
        key="Paperless"
        label="Paperless"
        value="paperless"
      />
    </WithStyles(ForwardRef(Tabs))>
  </WithStyles(ForwardRef(AppBar))>
</Fragment>
`;
